source/indexes.txt

```diff
@@ -0,0 +1,278 @@
+   :start-after: start-single-field
+   :end-before: end-single-field
+   :language: php
+   :copyable:
+   :dedent:
+
+.. To learn more about single field indexes, see the
+.. :ref:`php-single-field-index` guide.
+
+Compound Index
+--------------
+
+The following example creates a compound index of two ascending indexes
+on the specified fields:
+
+.. literalinclude:: /includes/usage-examples/index-code-examples.php
+   :start-after: start-compound
+   :end-before: end-compound
+   :language: php
+   :copyable:
+   :dedent:
+
+.. To learn more about compound indexes, see the :ref:`php-compound-index` guide.
+
+Multikey Index
+--------------
+
+The following example creates an ascending multikey index on the
+specified array-valued field:
+
+.. literalinclude:: /includes/usage-examples/index-code-examples.php
+   :start-after: start-multikey
+   :end-before: end-multikey
+   :language: php
+   :copyable:
+   :dedent:
+
+.. TODO To learn more about multikey indexes, see the :ref:`php-multikey-index`
+.. guide.
+
+Geospatial Index
+----------------
+
+The following example creates a 2dsphere index on the specified field
+that has GeoJSON object values:
+
+.. literalinclude:: /includes/usage-examples/index-code-examples.php
+   :start-after: start-geo
+   :end-before: end-geo
+   :language: php
+   :copyable:
+   :dedent:
+
+.. TODO: To learn more about geospatial indexes, see the :ref:`php-geospatial-index`
+.. guide.
+
+Unique Index
+------------
+
+The following example creates an ascending unique index on the specified
+field:
+
+.. literalinclude:: /includes/usage-examples/index-code-examples.php
+   :start-after: start-unique
+   :end-before: end-unique
+   :language: php
+   :copyable:
+   :dedent:
+
+.. TODO: To learn more about unique indexes, see the :ref:`php-unique-index`
+.. guide.
+
+Wildcard Index
+--------------
+
+The following example creates an ascending wildcard index on the
+collection:
+
+.. literalinclude:: /includes/usage-examples/index-code-examples.php
+   :start-after: start-wildcard
+   :end-before: end-wildcard
+   :language: php
+   :copyable:
+   :dedent:
+
+.. TODO: To learn more about wildcard indexes, see the :ref:`php-wildcard-index`
+.. guide.
+
+Clustered Index
+---------------
+
+You can create a clustered index when creating a new collection in a
+specified database. The following example creates a new collection with an
+ascending clustered index on the ``_id`` field:
+
+.. literalinclude:: /includes/usage-examples/index-code-examples.php
+   :start-after: start-clustered
+   :end-before: end-clustered
+   :language: php
+   :copyable:
+   :dedent:
+
+.. TODO: To learn more about clustered indexes, see the :ref:`php-clustered-index`
+.. guide.
+
+Atlas Search Index Management
+-----------------------------
+
+The following sections contain code examples that describe how to manage
+:atlas:`Atlas Search indexes </atlas-search/manage-indexes/>`. 
```

Comment:
Not sure if this is better addressed in some common template for driver docs on Atlas Search Index Management, but this seems like a good place to note that all of the driver APIs for search indexes are async. The driver methods themselves may be sync, but the main difference is that the server returns its response immediately while delegating the index management to happen in the background. This is why some of the existing driver example code elsewhere uses `listSearchIndexes()` to determine when the index is actually created and ready to use after initially invoking `createSearchIndex()` (or `createSearchIndexes()` in the case of a batch creation).

Reply:
i'll add a note! but yes, this can be a point made more clear in a dedicated page